airflow/api/common/experimental/mark_tasks.py

```diff
@@ -337,6 +340,9 @@ def set_dag_run_state_to_failed(dag, execution_date, commit=False, session=None)
         task.dag = dag
         tasks.append(task)
 
+    if not tasks:
+        return []
+
```

Comment:
This check may not be necessary as it's already handled inside `set_state()`?

Reply:
Probably not needed, you're right